org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/ScannablePositioner.java

```diff
@@ -115,29 +115,58 @@ public IPosition getPosition() throws ScanningException {
 
 	private final class MoveTask implements Callable<IPosition> {
 
-		@SuppressWarnings("rawtypes")
-		private IScannable scannable;
-		private IPosition  position;
+		private IScannable<?> scannable;
+		private IPosition     position;
 
 		public MoveTask(IScannable<?> iScannable, IPosition position) {
 			this.scannable = iScannable;
 			this.position  = position;
 		}
 
-		@SuppressWarnings("unchecked")
 		@Override
 		public IPosition call() throws Exception {
 			
 			// Get the value in this position, may be null for monitors.
-			Object value = position.get(scannable.getName());
+			Object value    = position.get(scannable.getName());
+			Object achieved = value;
 			try {
-			    scannable.setPosition(value, position);
+				achieved = setPosition(scannable, value, position);
 			    
 			} catch (Exception ne) {
 				abort(scannable, value, position, ne);
 				throw ne;
 			}
-			return new MapPosition(scannable.getName(), position.getIndex(scannable.getName()), scannable.getPosition()); // Might not be exactly what we moved to
+			// achieved might not be equal to demand
+			if (achieved == null) achieved = scannable.getPosition();
+			return new MapPosition(scannable.getName(), position.getIndex(scannable.getName()), achieved); 
+		}
+
+		@SuppressWarnings({ "rawtypes", "unchecked" })
+		private Object setPosition(IScannable scannable, Object value, IPosition position) throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)